### PR TITLE
feat: add auto tag and release workflow

### PR DESCRIPTION
**Pull request by newborn22**
## Checklist

-   [ ] Tests were added or are not required
-   [ ] Documentation was added or is not required

### CODE REVIEW

**Comment by newborn22** [comment]
🤖 says: comment test

**Comment by newborn22** [comment]
/pick-and-release feat/wescale_wesql_cluster_workflow 0.3.1-alpha22

**Comment by newborn22** [comment]
/pick-and-release feat/wescale_wesql_cluster_workflow 0.3.1-alpha22

**Comment by newborn22** [comment]
/pick-and-release feat/wescale_wesql_cluster_workflow 0.3.1-alpha22

File: .github/workflows/pick-and-release.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+          if git tag | grep -q "^$TAG_NAME$"; then
+            git tag -d $TAG_NAME
+            git push origin :refs/tags/$TAG_NAME
+          fi
+          
+          git tag $TAG_NAME
+          git push origin $TAG_NAME
+          
+          gh workflow run release-image.yml --ref $TAG_NAME$ -f image_tag=$TAG_NAME$
+          gh pr comment ${{ github.event.issue.number }} --body "🤖 says: tag is added, and image release starts running."
```
Comment (earayu) [inline .github/workflows/pick-and-release.yaml:46]: I believe the workflow would be clearer if 'release-image' and 'tag' were separated into two distinct steps.

Comment (earayu) [reply .github/workflows/pick-and-release.yaml:46]: Will be easier to debug.

File: .github/workflows/pick-and-release.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      - uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          fetch-depth: 0
+
+      - name: Automatic Cherry Pick
+        uses: vendoo/gha-cherry-pick@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
```
[the remark this answers is not in the document]
Comment (newborn22) [reply .github/workflows/pick-and-release.yaml:24]: if one of the steps fails, the following steps will be ignored by default.

File: .github/workflows/pick-and-release.yaml (hunk 1 above)
Comment (newborn22) [reply .github/workflows/pick-and-release.yaml]: fix

File: .github/workflows/pick-and-release.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            exit 1
+          fi
+          
+          git fetch origin $TARGET_BRANCH
+          git checkout $TARGET_BRANCH
+          
+          if git tag | grep -q "^$TAG_NAME$"; then
+            git tag -d $TAG_NAME
+            git push origin :refs/tags/$TAG_NAME
+          fi
```
[the remark this answers is not in the document]
Comment (newborn22) [reply .github/workflows/pick-and-release.yaml:47]: fix

**Review by earayu — COMMENTED** [review]
LGTM

**Review by earayu — APPROVED** [review]

**Pull request merged**